public/app/features/dashboard-scene/scene/row-actions/RowActions.tsx

```diff
@@ -0,0 +1,219 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import {
+  SceneComponentProps,
+  SceneGridItem,
+  SceneGridLayout,
+  SceneGridRow,
+  SceneObjectBase,
+  SceneObjectState,
+  SceneQueryRunner,
+  VizPanel,
+} from '@grafana/scenes';
+import { Icon, TextLink, useStyles2 } from '@grafana/ui';
+import appEvents from 'app/core/app_events';
+import { SHARED_DASHBOARD_QUERY } from 'app/plugins/datasource/dashboard';
+import { ShowConfirmModalEvent } from 'app/types/events';
+
+import { getDashboardSceneFor } from '../../utils/utils';
+import { DashboardScene } from '../DashboardScene';
+import { RowRepeaterBehavior } from '../RowRepeaterBehavior';
+
+import { RowOptionsButton } from './RowOptionsButton';
+
+export interface RowActionsState extends SceneObjectState {}
+
+export class RowActions extends SceneObjectBase<RowActionsState> {
+  public constructor(state: RowActionsState) {
+    super({
+      ...state,
+    });
+  }
+
+  private get _dashboard(): DashboardScene {
+    return getDashboardSceneFor(this);
+  }
+
+  private updateLayout(rowClone: SceneGridRow): void {
+    const row = this.getParent();
+
+    const layout = this._dashboard.state.body;
+
+    if (!(layout instanceof SceneGridLayout)) {
+      throw new Error('Layout is not a SceneGridLayout');
+    }
+
+    // remove the repeated rows
+    const children = layout.state.children.filter((child) => !child.state.key?.startsWith(`${row.state.key}-clone-`));
+
+    // get the index to replace later
+    const index = children.indexOf(row);
+
+    if (index === -1) {
+      throw new Error('Parent row not found in layout children');
+    }
+
+    // replace the row with the clone
+    layout.setState({
+      children: [...children.slice(0, index), rowClone, ...children.slice(index + 1)],
+    });
+  }
+
+  public getParent(): SceneGridRow {
+    if (!(this.parent instanceof SceneGridRow)) {
+      throw new Error('RowActions must have a SceneGridRow parent');
+    }
+
+    return this.parent;
+  }
+
+  public getDashboard(): DashboardScene {
+    return this._dashboard;
+  }
```

Comment:
Don't think we need the private getter `_dashboard` while we have a public `getDashboard` method.